base/show.jl

```diff
@@ -425,6 +425,18 @@ function show(io::IO, @nospecialize(x::Type))
         show_datatype(io, x)
         return
     elseif x isa Union
+        if x <: StridedArray && x.a <: DenseArray
+            T, N = eltype(x.a), ndims(x.a)
```

Comment:
I think this `ndims` call will fail if `x.a` is an abstract type.

Reply:
Ah, good point. What if I just add an `applicable(ndims, x.a)` to the condition above?

Reply:
Fixed by explicitly ensuring that `x.a` is explicitly a `DenseArray` and not just a subtype on the previous line with:

```Julia
if x.a isa DataType && Core.Compiler.typename(x.a) === Core.Compiler.typename(DenseArray)
    T, N = x.a.parameters
```

Then I can dig into `.parameters` and know they mean `T, N`.